src/material-experimental/mdc-slider/slider.ts

```diff
@@ -342,19 +345,28 @@ export class MatSlider implements AfterViewInit, OnChanges, OnDestroy, ControlVa
 
   ngAfterViewInit() {
     this._isInitialized = true;
-    this._foundation.init();
 
-    // The standard Angular Material slider is always using discrete values. We always
-    // want to enable discrete values and support ticks, but want to still provide
-    // non-discrete slider visual looks if thumb label is disabled.
-    // TODO(devversion): check if we can get a public API for this.
-    // Tracked with: https://github.com/material-components/material-components-web/issues/5020
-    (this._foundation as any).isDiscrete_ = true;
+    if (this._platform.isBrowser) {
+      // The MDC slider foundation accesses DOM globals, so we cannot initialize the
+      // foundation on the server. The foundation would be needed to move the thumb
+      // to the proper position and to render the ticks.
+      this._foundation.init();
```

Comment:
@mmalerba @jelbourn Unfortunately there is an issue with MDC slider (and I guess it can be for other MDC foundations too). The slider thumb is moved in the MDC foundation through calculated pixels. This requires the foundation to perform element measurements using `getBoundingClientRect`.

In the server though, `getBoundingClientRect` is not available and therefore the foundation cannot move the thumb to the proper location. This is an issue for server-side rendering because the pre-rendered page will have sliders that do not reflect the initial value (until the page is hydrated)

Reply:
Yeah I would create an issue on their repo. They've said that they want all of their components to be SSR compatible, so this sounds like a bug to me

Reply:
For reference: here is the issue: https://github.com/material-components/material-components-web/issues/5058